tests/console/yast2_vnc.pm

```diff
@@ -46,13 +46,14 @@ sub configure_remote_admin {
 
 sub check_service_listening {
     my $cmd_check_port = $is_older_product ? 'netstat' : 'ss -tln | grep -E LISTEN.*:5901';
-    if (script_run($cmd_check_port)) {
-        record_soft_failure 'boo#1088646 - service vncmanager is not started automatically';
-        systemctl('status vncmanager', expect_false => 1);
-        systemctl('restart vncmanager');
-        systemctl('status vncmanager');
-        assert_script_run $cmd_check_port;
+    my $is_open;
+    for (0 .. 5) {    # try several times
+        $is_open = 1 if !script_run($cmd_check_port);
```

Comment:
maybe unless is more perlish here. However i would like to point you to `script_retry` from the `utils` module. using it will avoid this loop, sleep and die calls here.

Reply:
Ah, I was looking for it in testapi, then could not find it there, obvioulsy. Thanks.